onward/app/weather/WeatherApi.scala

```diff
@@ -87,3 +89,18 @@ class WeatherApi(wsClient: WSClient, context: ApplicationContext)(implicit ec: E
       Json.fromJson[Seq[ForecastResponse]](r).get
     })
 }
+
+object WeatherApi extends Logging {
+
+  def retryWeatherRequest(request: () => Future[JsValue], retryDelay: FiniteDuration, scheduler: Scheduler, attempts: Int)(implicit ec: ExecutionContext): Future[JsValue] = {
+    def loop(attemptsRemaining: Int): Future[JsValue] = {
+      request().recoverWith {
+        case NonFatal(error) =>
+          log.warn(s"Error on weather request, attempts remaining $attemptsRemaining", error)
```

Comment:
I don't think we want to log every failure. Logging the last one should be enough. What do you think?